CONTRIBUTING.md

```diff
@@ -9,23 +9,32 @@ There are 4 main ways of contributing to the PyMC project (in descending order o
 * Contributing or improving the documentation (`docs`) or examples (`pymc/examples`)
 * Submitting issues related to bugs or desired enhancements
 
-# Opening issues
+## Opening issues
 
 We appreciate being notified of problems with the existing PyMC code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-devs/pymc/issues), rather than on social media or by direct email to the developers.
 
 Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
 Filter on the ["beginner friendly"](https://github.com/pymc-devs/pymc/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) label for issues which are good for new contributors.
 
-# Contributing code via pull requests
+## Etiquiette for code contributions
+When you start working working on an issue, open a `Draft` pull request as soon as you're doing the first steps.
+Make sure to mention the issue number in the PR description or commit message.
+This way others can see that the issue is being worked on.
+
+Comments like "I want to work on this issue" will be ignored.
+This is because we value getting things done over pledges and found that waiting weeks for follow-ups to such pledges slows down the progress.
+Similarly, we may decide to close or take over pull requests where the author is not responsive.
```

Comment:
I agree with both comments, the language is harsh and I'm not confident in the proposed process.

I would just instead stay, any issue without an open pull request is available for work. However if your pull request is more than two weeks old with no activity and someone else expresses interest reserve the right to either close it, or let someone else take it over.